CharlieBackend.Api/Controllers/CoursesController.cs

```diff
@@ -46,27 +48,32 @@ public async Task<ActionResult> PostCourse(CreateCourseModel courseModel)
 
         [Authorize(Roles = "2, 4")]
         [HttpGet]
-        public async Task<ActionResult<List<CourseModel>>> GetAllCourses()
+        public async Task<ActionResult<IList<CourseDto>>> GetAllCourses()
         {
 
-            var courses = await _coursesService.GetAllCoursesAsync();
+            var courses =  await _coursesService.GetAllCoursesAsync();
+
+            var dtoResult = new List<CourseDto>();
+
+            foreach (var item in courses)
+            {
+                dtoResult.Add(_mapper.Map<CourseDto>(item));
+            }
```

Comment:
You don't need to use mapper in controller. As suggested in previous comments, please leave only dto or model